### ISSUE TITLE
Retire plugin

### ISSUE THREAD

**Issue by slawekjaranowski** [s4u/sign-maven-plugin#195]
As [Apache Maven GPG Plugin](https://maven.apache.org/plugins/maven-gpg-plugin/) supports now:

 - support BC BouncyCastle java library signer -gpg binary is not necessary
 - support environment variables
 - works with Maven 4
 
https://maven.apache.org/plugins/maven-gpg-plugin/examples/deploy-signed-artifacts.html#sign-using-bc-signer

I'm going to retire this plugin

Task to do:
- [x]  prepare migration example to maven-gpg-plugin
- [x]  add info about retired in docs
- [x] deprecate Mojo with info about retiring
- [x]  release latest version
- [x] make repository read-only

**Comment by mkarg** [comment]
Farewell, sign plugin!

OTOH I thought your intention for starting this plugin was to get rid of a the need for a locally installed GPG. Does that mean that Maven GPG Plugin works without local GPG now? 🤔

**Comment by slawekjaranowski** [comment]
> Farewell, sign plugin!
> 
> OTOH I thought your intention for starting this plugin was to get rid of a the need for a locally installed GPG. Does that mean that Maven GPG Plugin works without local GPG now? 🤔

Exactly it was a intention - signing without additional binary and better support builds on CI system

Now we have the same in gpg-plugin thanks to BC - Bouncy Castle Java library additional binary is not necessary.
Also we have support for configuration keys by environments variables.

I will prepare some manual about migration. 

So maintain next plugin with the same feature is not justified.

**Comment by mkarg** [comment]
So in fact it is not GPG plugin anymore, but BC plugin. ;-)

**Comment by slawekjaranowski** [comment]
> So in fact it is not GPG plugin anymore, but BC plugin. ;-)

No. There is an option: https://maven.apache.org/plugins/maven-gpg-plugin/sign-mojo.html#signer

> The name of the Signer implementation to use. Accepted values are "gpg" (the default, uses GnuPG executable) and "bc" (uses Bouncy Castle pure Java signer).

**Comment by slawekjaranowski** [comment]
The plugin is retired: the final release 1.2.0 is out, the [migration guide](https://www.simplify4u.org/sign-maven-plugin/migration.html) to maven-gpg-plugin is published, CI and Dependabot are removed, and the repository is being archived.
